src/resource.c

```diff
@@ -805,7 +805,8 @@ coap_notify_observers(coap_context_t *context, coap_resource_t *r) {
       /* TODO: do not send response and remove observer when
        *  COAP_RESPONSE_CLASS(response->hdr->code) > 2
        */
-      if (response->type == COAP_MESSAGE_CON) {
+      if (response->type == COAP_MESSAGE_CON ||
+          (r->flags & COAP_RESOURCE_FLAGS_NOTIFY_NON_ALWAYS)) {
```

Comment:
I think that always resetting `non_cnt` to 0 is the correct approach here. But we may also want to add the special treatment of `COAP_RESOURCE_FLAGS_NOTIFY_NON_ALWAYS` to the if-clause in line 797 (where `response->type` is set) to avoid surprises in case libcoap is built with `-DCOAP_OBS_MAX_NON=0`.

Reply:
Good point.  Special treatment added in and code pushed.